src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -282,8 +282,13 @@ public void clear()
           if (!iterator.hasNext()) {
             int size = tasks.size();
             if (size > 0) {
+              Runnable task;
               do {
-                tasks.pollUnsafe().run();
+                task = tasks.poll();
+                if (task == null) {
+                  break;
+                }
+                task.run();
               }
               while (--size > 0);
```

Comment:
That would be a good way to balance it but it is also an arbitrary limit isn't it. I am fine with it.